### PR TITLE
Replace versioned links with /stable/

### LINKED ISSUE #9701: Update versioned links to use /stable/ in CC
When cross-referencing CRDB docs, CockroachCloud docs should use the `/stable/` versions of links instead of `/vX.X/`. This way, links do not have to be manually updated whenever the `stable` version is updated.

Note: Some time after the v20.2 release, the `/stable/` alias seemed to stop working in TeamCity. _Existing_ links using `/stable/` still seem to work, but any _new_ links throw an error in the build log.

### PR DESCRIPTION
**Pull request by lnhsingh**
Closes #9701

### CODE REVIEW

**Comment by lnhsingh** [comment]
@jseldess - adding you as a reviewer on this, hoping you can help figure out why new instances of `/stable/` seem to break the TC build. _Existing_ links using `/stable/` still seem to work, but any _new_ links throw an error in the build log. But I'm also seeing in this TC build that other links are breaking too, when I don't think they should be.

**Comment by lnhsingh** [comment]
It seems like a lot of pages aren't being built for some reason, for example:
- http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f315daf9152aa7c1426c0828cf81f56be6d68082/cockroachcloud/user-authorization.html
- http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f315daf9152aa7c1426c0828cf81f56be6d68082/cockroachcloud/free-faqs.html
- http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f315daf9152aa7c1426c0828cf81f56be6d68082/cockroachcloud/quickstart-trial-cluster.html

Which would explain why the links to these pages break, but I'm not sure what's preventing them from being built.

And then on pages that _say_ that the `stable` link is broken, that doesn't seem to be the case. For example, if you go to http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f315daf9152aa7c1426c0828cf81f56be6d68082/cockroachcloud/learn-cockroachdb-sql.html#show-databases, `shell commands` is linked to http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f315daf9152aa7c1426c0828cf81f56be6d68082/stable/cockroach-sql.html#commands. TC is listing this link as `missing reference`, but when I click the link, it works.

File: cockroachcloud/stream-changefeed-to-snowflake-aws.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
----
+../stable---
```
Comment (jseldess) [inline cockroachcloud/stream-changefeed-to-snowflake-aws.md:1]: This looks like a mistake?

File: cockroachcloud/user-authorization.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
----
+../stable---
```
Comment (jseldess) [inline cockroachcloud/user-authorization.md:1]: Mistake?

File: cockroachcloud/free-faqs.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
----
+../stable---
```
Comment (jseldess) [inline cockroachcloud/free-faqs.md:1]: Mistake?

File: cockroachcloud/quickstart-trial-cluster.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
----
+../stable---
```
Comment (jseldess) [inline cockroachcloud/quickstart-trial-cluster.md:1]: Mistake?

File: cockroachcloud/build-a-python-app-with-cockroachdb-django.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
----
+../stable---
```
Comment (jseldess) [inline cockroachcloud/build-a-python-app-with-cockroachdb-django.md:1]: Mistake?

File: cockroachcloud/connect-to-a-free-cluster.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
----
+../stable---
```
Comment (jseldess) [inline cockroachcloud/connect-to-a-free-cluster.md:1]: Mistake?

**Comment by jseldess** [comment]
@lnhsingh, it looks like you accidentally messed up the front-matter of a bunch of pages. Let's see if the problem goes away after fixing those files.

**Comment by lnhsingh** [comment]
Fixing the front matter fixed some of the errors. There are still errors from `/stable` that don't seem to be errors though. For example, on [FAQs](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b2911e378696c8ca08a125c3ba90cb9c5dc6ddf0/cockroachcloud/free-faqs.html#can-i-backup-my-cockroachcloud-free-beta-cluster-does-cockroach-labs-take-backups-of-my-cluster), TC is saying the last link on the page (to `../stable/cockroach-sql.html#general`) is `missing reference`, but when I click it, it works.

**Comment by jseldess** [comment]
Hmm, yeah, something strange is happening there that I don't understand. I vaguely remember not being able to use `/stable` in CC docs links before. When we leave those links with `/v20.2`, they work fine? If so, I'd say we should do that for now, add an item to the [GA Release Checklist](https://cockroachlabs.atlassian.net/wiki/spaces/PROD/pages/402718726/GA+Release+Checklist) to update these to the new stable when we GA a new version, and open a docs issue to figure out why this isn't working. Once YZ is back, she can help us figure that out.

**Comment by lnhsingh** [comment]
I was able to add `/stable` links in before-- I think the issue might've
started after the v20.2 release. Yes, `v20.2` links work fine, which is
what I've been doing so I can just scrap this PR for now

**Comment by jseldess** [comment]
@lnhsingh, can you track down an earlier repo state/PR where stable was working in CC doc links?

**Comment by lnhsingh** [comment]
I looked and can't find an exact PR (I can try looking more later), but I
did update some of the links to use `/stable` and those are fine. It seems
like only newly added `/stable` links throw errors, but I'm not sure

**Pull request closed without merging**